python/tests/test_pyvw.py

```diff
@@ -259,40 +274,54 @@ def test_multiclass_label():
     assert str(ml) == '2:0.2'
 
 def test_multiclass_label_example():
-    vw_ex = vw(quiet=True)
-    ex = vw_ex.example('1 |a two features |b more features here')
+    n = 4
+    model = pyvw.vw(loss_function='logistic', oaa=n, quiet=True)
+    ex = model.example('1 | a b c d', 2)
     ml2 = pyvw.multiclass_label(ex)
-    assert ml2.label == 1065353216
+    assert ml2.label == 1
     assert ml2.weight == 1.0
     assert ml2.prediction == 0
-    assert str(ml2) == '1065353216'
+    assert str(ml2) == '1'
+
+
+class TestExample:
```

Comment:
Correct me if I'm wrong, but it looks like several functions in this class have side effects. Which means the order in which they are called will affect the test results. If that's the case, then the behavior is wrong for unit tests. Each function call should be completely independent from any others, and any requisite state information needs to be explicitly passed in as parameters.

Reply:
Should it be one complete test as in earlier commit?

Reply:
Its better to have many small, independent functions that tests a specific piece of functionality. If thats not possible, then one large function may be unavoidable. However, it should never be assumed that tests will be called in a specific order; in fact some testing frameworks don't run tests in the order they appear in code

Reply:
I agree. But for the tests of example with this case, it seems good to have it one test. Is this good idea?

Reply:
A single function for `test_example` is far too general; there are a lot of components to an example and there's no way to tell what's actually being tested.

Performance in the testing code (so anything in `test_pyvw.py`) is not really a concern. If you can break the unit tests into smaller pieces by doing duplicate work (like setting up the same example in different functions), thats fine.

If there are multiple components you feel are functionally the same thing, and it makes more sense to keep the checks in the same function call, then please do. We may ask why in some cases, but as long as there's a reasonable justification for the decision, it should be fine.